### PR TITLE
Rightclick option to random play song

### PR DESCRIPTION
**Pull request by Crystalwarrior**
This is a really nice function to just play a random song from your currently visible music list. Respects the search string!

### CODE REVIEW

**Comment by oldmud0** [comment]
are you testin' me or somethin

**Comment by Crystalwarrior** [comment]
> 
> 
> are you testin' me or somethin

:)

File: src/courtroom.cpp [hunk 1]
```diff
@@ -4028,6 +4031,18 @@ void Courtroom::music_synchronize(bool toggle)
     music_flags &= ~SYNC_POS;
 }
 
+void Courtroom::music_random()
+{
+  QList<QTreeWidgetItem *> clist;
+  QTreeWidgetItemIterator it(ui_music_list, QTreeWidgetItemIterator::NotHidden | QTreeWidgetItemIterator::NoChildren);
```
Comment (oldmud0) [inline src/courtroom.cpp:4037]: Not clear how this avoids selecting categories.

Comment (Crystalwarrior) [reply src/courtroom.cpp:4037]: I explained it, it treats a category as an entry with any children. No children = not a category.

**Comment by oldmud0** [comment]
this will most assuredly break when subcategories are introduced

**Pull request merged**